v3_async_wip/v3_async_wip/custom_typing.py

```diff
@@ -0,0 +1,16 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import Union, Dict, List, Tuple, Any
+
+# typing does not support recursion, so unfortunately, we have to use Any for elements of
```

Comment:
>typing does not support recursion, so [](http://example.com/codeflow?start=2&length=37)

Could you use a forward reference? I don't think it would fix all problems, but it would make the type more accurate.  [https://peps.python.org/pep-0484/#forward-references](https://peps.python.org/pep-0484/#forward-references)